### PR TITLE
modify gui-docker and add documentation for how to use docker

### PR DESCRIPTION
**Pull request by ipa-rwu**

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
Should we be looking at [osrf/rocker](https://github.com/osrf/rocker) for this?

It's not perfect, but does offer (quite) some of the functionality this PR appears to introduce/document.

**Comment by ipa-rwu** [comment]
> Should we be looking at [osrf/rocker](https://github.com/osrf/rocker) for this?
> 
> It's not perfect, but does offer (quite) some of the functionality this PR appears to introduce/document.

Sure. It is also mentioned in [ROS Wiki](http://wiki.ros.org/docker/Tutorials/Hardware%20Acceleration).

File: gui-docker [hunk 1]
```diff
@@ -0,0 +1,152 @@
+#!/bin/bash -u
+
```
Comment (ipa-hsd) [inline gui-docker:2]: I think we need to specify the source of the script
https://github.com/ros-planning/moveit2/blob/main/.docker/gui-docker

File: docs/How to start with docker GUI copy.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+# Start with docker
```
Comment (ipa-hsd) [inline docs/How to start with docker GUI copy.md:1]: This doc has been added twice? It says "copy" in the doc name

File: gui-docker (hunk 1 above)
Comment (ipa-hsd) [reply gui-docker:2]: Or does it makes sense to send a PR to the moveit2 repo directly?

Comment (ipa-rwu) [reply gui-docker:2]: I am not sure. The main different is how to check if there is nvidia card.

moveit2 use `lspci | grep -i "vga .*nvidia"`. But it doesn't work for me.

So I changed to `lspci | grep -i "nvidia" `

**Comment by ipa-hsd** [comment]
> Should we be looking at osrf/rocker for this?

We didn't get much time to test it for the November training. We tested this solution and works fine. But we open to test it in the future. Thanks for the suggestion!

**Pull request merged**